ethers-etherscan/src/contract.rs

```diff
@@ -284,12 +284,28 @@ impl Client {
     /// # }
     /// ```
     pub async fn contract_abi(&self, address: Address) -> Result<Abi> {
+        // apply caching
+        if let Some(ref cache) = self.cache {
+            if let Ok(Some(abi)) = cache.get_abi(address) {
+                return Ok(abi)
+            }
+        }
+
         let query = self.create_query("contract", "getabi", HashMap::from([("address", address)]));
         let resp: Response<String> = self.get_json(&query).await?;
+        if resp.result.starts_with("Max rate limit reached") {
```

Comment:
we have to dupe the check here cus it deserializes `message` as a string.. really annoying but no way around it really